### DISCUSSION TITLE
Questions on Co-assembly with metaSPAdes for Metagenomic Analysis

### DISCUSSION THREAD

**Discussion by Pompamm** [ablab/spades#1253] [Q&A]
Dear SPAdes users and professionals,

I'm seeking advice regarding co-assembly using metaSPAdes for metagenomic analysis. I intend to perform the co-assembly of three metagenome samples, which exhibit similar microbial populations, with the aim of binning and constructing Metagenome-Assembled Genomes (MAGs). I have two main questions:

1. Code Verification: After executing metaSPAdes with the provided code:
metaspades.py --pe1-1 Sample1_Forward_paired_q20-100.fq.gz --pe1-2 Sample1_Reverse_paired_q20-100.fq.gz \
        --pe2-1 Sample2_Forward_paired_q20-100.fq.gz --pe2-2 Sample2_Reverse_paired_q20-100.fq.gz \
        --pe3-1 Sample3_Forward_paired_q20-100.fq.gz --pe3-2 Sample3_Reverse_paired_q20-100.fq.gz \
        -k 77,91 -t 20 -o ../co-assembly --meta

The result came out, "The program was terminated by segmentation fault" error. Please verify if the code is correct or suggest any potential corrections.
[co-assembly_mtg_sbi2_log.txt](https://github.com/ablab/spades/files/14460203/co-assembly_mtg_sbi2_log.txt)

2. Co-assembly Approach: Should I concatenate the reads of the three samples (concatenating forward reads and reverse reads separately) and then perform co-assembly? I noticed that metaSPAdes typically works with a single library, hence my uncertainty about the optimal approach.

Additional context:
- I aim to compare the completeness of MAGs and the number of bins obtained from both individual assembly and co-assembly.
- I'm open to suggestions for alternative assembly tools that might be more compatible with co-assembly, if necessary.

Thank you in advance for your insights and suggestions. Your input is greatly appreciated.

**Comment by asl** [comment]
> The result came out, "The program was terminated by segmentation fault" error. Please verify if the code is correct or suggest any potential corrections.
> [co-assembly_mtg_sbi2_log.txt](https://github.com/ablab/spades/files/14460203/co-assembly_mtg_sbi2_log.txt)

The log reads:
```
  5:05:01.581    41G / 60G   ERROR   General                 (mmapped_reader.hpp        :  52)   mmap(2) failed. Reason: Cannot allocate memory. Error code: 12
```
So you're running out of RAM.

> Co-assembly Approach: Should I concatenate the reads of the three samples (concatenating forward reads and reverse reads separately) and then perform co-assembly? I noticed that metaSPAdes typically works with a single library, hence my uncertainty about the optimal approach.

You need to pass them as a single library.
